sdv/metadata/multi_table.py

```diff
@@ -282,6 +304,9 @@ def add_relationship(
         self._validate_relationship_does_not_exist(
             parent_table_name, parent_primary_key, child_table_name, child_foreign_key
         )
+        self._validate_foreign_key_is_not_reused(
+            parent_table_name, parent_primary_key, child_table_name, child_foreign_key
+        )
```

Comment:
I don't think we need to call `_validate_foreign_key_is_not_reused` again here because we already called `_validate_relationships`. Also do you know if there's a specific reason why `_validate_relationship_does_not_exist` isn't called in `_validate_relationships`?

Reply:
Yes, because we use it during `validate` which validates each relationship one by one. If we called this `_validate_relationship_does_not_exist` this would raise errors saying that the relationships exist.